File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_126/javascript/solution.js
function is_sorted(lst) {
  const counts = {};
  
  for (let i = 0; i < lst.length; i++) {
    const current = lst[i];
    
    if (i > 0 && lst[i - 1] > current) {
      return false;
    }
    
    counts[current] = (counts[current] || 0) + 1;
    
    if (counts[current] > 2) {
      return false;
    }
  }
  
  return true;
}

File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_126/java/solution.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static boolean isSorted(ArrayList<Long> lst) {
        Map<Long, Integer> counts = new HashMap<>();
        Long previous = null;
        
        for (Long current : lst) {
            if (previous != null && previous > current) {
                return false;
            }
            
            int count = counts.getOrDefault(current, 0) + 1;
            if (count > 2) {
                return false;
            }
            counts.put(current, count);
            
            previous = current;
        }
        
        return true;
    }
}
